tests/components/enphase_envoy/test_binary_sensor.py

```diff
@@ -0,0 +1,101 @@
+"""Test Enphase Envoy binary sensors."""
+
+from unittest.mock import AsyncMock
+
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.enphase_envoy.const import Platform
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from tests.common import MockConfigEntry
+from tests.components.enphase_envoy import setup_with_selected_platforms
+from tests.components.enphase_envoy.conftest import ALL_FIXTURES, BINARY_FIXTURES
+
+
+@pytest.mark.parametrize(
+    ("mock_envoy", "entity_count"), *BINARY_FIXTURES, indirect=["mock_envoy"]
+)
+async def test_binary_sensor(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    snapshot: SnapshotAssertion,
+    mock_envoy: AsyncMock,
+    entity_registry: er.EntityRegistry,
+    entity_count: int,
+) -> None:
+    """Test enphase_envoy binary_sensor entities."""
+    await setup_with_selected_platforms(hass, config_entry, [Platform.BINARY_SENSOR])
+
+    # number entities states should be created from test data
+    assert len(hass.states.async_all()) == entity_count
+
+    entity_entries = er.async_entries_for_config_entry(
+        entity_registry, config_entry.entry_id
+    )
+
+    if entity_count == 0:
+        assert len(entity_entries) == 0
+    else:
+        # compare registered entities against snapshot of prior run
+        assert entity_entries
+        for entity_entry in entity_entries:
+            assert entity_entry == snapshot(name=f"{entity_entry.entity_id}-entry")
+            assert hass.states.get(entity_entry.entity_id) == snapshot(
+                name=f"{entity_entry.entity_id}-state"
+            )
```

Comment:
We try to avoid using `if` statements in tests as much as possible as it makes it much harder to read and see what we are actually testing

Reply:
Removed where possible.